src/forgeAPI/java/net/neoforged/api/distmarker/Dist.java

```diff
@@ -29,6 +29,48 @@
  *     <li>{@link #DEDICATED_SERVER} is the <em>dedicated server</em> distribution,
  *     it contains a server, which can simulate the world and communicates via network.</li>
  * </ul>
+ * <p>
+ * When working with Dist-specific code, it is important to guard invocations such that
+ * classes invalid for the current Dist are not loaded.
+ * <p>
+ * This is done by obeying the following rules:<br>
+ * 1. All Dist-specific code must go in a separate class.<br>
+ * 2. All accesses to the Dist-specific class must be guarded by a Dist check.
+ * <p>
+ * Following these rules ensures that a Dist-induced classloading error will never occur.
+ * <p>
+ * An example of these rules in action is shown below:
+ * <p>
+ * <code><pre>
```

Comment:
I lament the fact that MergeTool isn't on Java 17, to make use of the `{@snippet}` javadoc tag.